sdk/state/update.go

```diff
@@ -18,16 +18,24 @@ type Payment struct {
 	FromInitiator         bool
 }
 
+func (p Payment) isEqual(p2 Payment) bool {
```

Comment:
I find the `isEqual` function name a bit misleading here, since you're not fully comparing the objects (and you're actually expecting the signatures to be different). Could we either:

- Use a different name, such as `isEquivalent`? or
- Document why we're not checking for all fields in a comment?

Reply:
good point. `isEquivalent` makes sense to me. I also added a comment that hopefully makes it clearer